### DISCUSSION TITLE
Library Selection Method - CDE 6347743 v2

### DISCUSSION THREAD

**Discussion by GeoffLyle** [CBIIT/ccdi-federation-api#47] [Metadata Field Ratification]
~~caDSR Link: [6347743](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6347743%20and%20ver_nr=1)~~
caDSR Link: [6347743 v2](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6347743%20and%20ver_nr=2)

Definition: "The type of systematic actions performed to select or enrich DNA fragments used in analysis by high-throughput sequencing."

Issues with CDE:
The current definition for permissible value **PolyA** does not fit the CDE definition - "1. The process of formation of poly(adenylic acid). 2. The covalent modification of a macromolecule (e.g., mRNA) by the formation of a polyadenylyl moiety covalently linked to the macromolecule."  Federation members think the term 'PolyA' would describe their library selection method when it is unknown where 'Oligo-DT' or 'Poly-T Enrichment' was used to generate the sequencing library.

Proposed Solutions:

- Change the definition of PolyA (ex. “A genomic library where the sample has been enriched in polyadenylated RNA transcripts prior to amplification”.) - Preferred​
- Request new PV (PolyA Selection)

**Comment by GeoffLyle** [comment]
Slack Discussion:

_From Clay McLeod on October 25th, 2023_ - For [CDE 6347743](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6347743%20and%20ver_nr=1) ("Sequencing Library Creation Method Type"), some permissible values are incredibly specific and include the experimental details of how the library was selected (e.g., "Oligo-dT" versus "Poly-T Enrichment", which have some fine-grained experimental differences, but have the same intended effect of mature messenger RNA isolation), some permissible values capture the general idea of how the library was selected (e.g., "rRNA depletion", which I interpret as encompassing a broad family of experimental techniques that does not capture the aforementioned level of experimental detail, such as whether mtRNA is removed during ribosomal depletion), and the permissible values that are just incredibly general (e.g., "Affinity Enrichment", which describes itself as simply "Any of various techniques designed to select or increase a target item in a mixed sample."—so technically many of the permissible values listed also match this as well). Further, though I personally do consider the concepts of molecule isolation/enrichment and size selection to fall under the general umbrella of the CDE name ("Sequencing Library Creation Method Type"), these concepts are orthogonal at the level of granularity they are being described in the permissible values—for example, one performs both an enrichment step and a size selection step for many of the available protocols. In this case, do we allow multiple values for the CDE?

_From Geoff Lyle on October 26th, 2023_ - Thank you for looking further into these CDEs, my initial review of [CDE 6347743](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6347743%20and%20ver_nr=1) ("Sequencing Library Creation Method Type") did not catch the differences between the "Oligo-dT" and "Poly-T Enrichment" permissible values.  Our team met today to discuss these points you raised.
One quick aside for this CDE (6347743), there is a permissible value "PolyA" with the long name of "RNA Polyadenylation" and a definition "1. The process of formation of poly(adenylic acid). 2. The covalent modification of a macromolecule (e.g., mRNA) by the formation of a polyadenylyl moiety covalently linked to the macromolecule."  Our group was unaware of "PolyA" being used with this definition when discussing the method by which a RNA-Seq sequencing library was generated, but perhaps others may be more familiar with this.

> Is the intent to always to pick the most specific value rather than select all matching values?
> Do we allow multiple values for all of these CDEs listed?

This was one of the questions at the first data harmonization meeting when we brought up how to harmonize the CDE regarding Race ([CDE 2192199](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=2192199%20and%20ver_nr=1)).  It was suggested we allow multiple values for this field since the permissible values do not account for multiracial people.  Given this, we should be able to allow multiple values for other CDEs as well.  In this specific case (CDE 6347743), we could allow for multiple values to be selected, which would allow us to capture the specific as well as general terms for the data.  This is likely to cause inconsistencies across sites, but that is likely a given with Federated data unless we have some sort of oversight of the harmonized data being shared (unsure who this would be and how they would be funded).

> We report all of the samples as Oligo-dT (technically will not be true for a subset of samples that we cannot distinguish but functionally should be “equivalent”—rare biases due to the experimental details not withstanding).

We agree that this is a bad option, and not a path we would take.

> We report absolutely nothing about the library selection protocol for these samples, as we cannot be absolutely sure on the details.

Agreed that this is a less useful option for the users, yet we would take this path.  We are very wary of harmonizing to a specific value if we are not sure that value fits our data.  We would report a blank value and include the unharmonized field in the ancestor field (the harmonized fields currently have a way of linking back to the unharmonized data, so users could look there for our best knowledge of what value fits).

> We can introduce a new field that captures the level of granularity for information we are sure of (mRNA enrichment versus ribosomal depletion).

We would like to update this CDE to fit our data, so we will likely have to work with the NIH to determine if this is feasible.
**Summary**

- Based on determinations of prior CDEs, harmonized fields should be allowed to have _multiple_ values when necessary.
- If none of the permissible values in a CDE fit our data, we will return a blank field with a key linking back to the unharmonized data field we feel best fits the CDE.  In cases where we feel too much useful data is being lost (ex. [CDE 6347743](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6347743%20and%20ver_nr=1)) we would like to determine whether the CDE can be modified or whether a more fitting CDE exists.

_Geoff Lyle's data harmonization subgroup meeting notes on November 17th, 2023_ -
Proposed Solutions:

- Change the definition of permissible value 'PolyA' to "A genomic library where the sample has been enriched in polyadenylated RNA transcripts prior to amplification".  Changing this permissible value's definition would bring it more in line with the definition of the CDE, "The type of systematic actions performed to select or enrich DNA fragments used in analysis by high-throughput sequencing."
- If changing the definition of a permissible value is not allowed, then created a new permissible value "PolyA Selection" would be a second, less preferred solution.

**Comment by jknable** [comment]
We can revise the CDE PolyA definition to reflect the current NCI Thesaurus definition for PolyA: 
"The enzymatic addition of a sequence of adenylyl residues at the 3' end of an RNA molecule."

Would that work for your group?

**Reply by lydiabrogers** [reply]
In harmonization team meeting group asked the definition for PolyA to be “A genomic library where the sample has been enriched in polyadenylated RNA transcripts prior to amplification”.

**Comment by hbeale** [comment]
Slack discussion: 

Holly Beale
@allison
@cmcleod
 Can you or someone on your team put "used" in the columns of values you expect to use for Library Selection Method? I've done it for our team. [This](https://docs.google.com/spreadsheets/d/1eT2ZahJTUq0ZihhVouSSqKQCYO0z2KrFDbgNB06DC2w/edit#gid=0) is a google sheet anyone can edit. Let me know if this process should be more formal. (cc @Geoff Lyle)(@bfurner
 I don't think your group will use this CDE, but let me know if I'm wrong)

bfurner
we will not use that, correct


Geoff Lyle
adding 
@Hannah Calkins


ssandor
Added for St. Jude (
@cmcleod 
@jsunny
@jnoble42
)



Hannah Calkins
I have updated for CHOP/CBTN (edited) 


Holly Beale
Thanks for the quick action, everybody. Here are the final results. I had second thoughts about "Other", and removed Treehouse's suggestion for it. Unless I hear that you want to discuss it more, I'll send this list to Janice and ask her to see if the owners are up for deleting all other values.


Holly Beale
PolyA (CHOP, St Jude, Treehouse)
rRNA Depletion (CHOP, St Jude, Treehouse)
Hybrid Selection (CHOP, Treehouse)
PCR (St Jude)
Poly-T Enrichment (CHOP)
Random (St Jude)
Unspecified (St Jude)
Not applicable (St Jude)

**Comment by GeoffLyle** [comment]
Further discussion between @hbeale and @calkinsh led to agreement to using an umbrella term **Poly-A Enriched Genomic Library** to cover both **Poly-T Enrichment** and **Oligo-dT**.

@jknable Next steps for CDE:

1. **Poly-T Enrichment** to be changed to **Poly-A Enriched Genomic Library**
2. And drop **Oligo-dT**

**Comment by GeoffLyle** [comment]
At April 5th harmonization meeting, it was decided that the CDE would be accepted once **Oligo-dT** and **PolyA** are dropped as Permissible Values and **Poly-T Enrichment** is changed to **Poly-A Enriched Genomic Library**.  Term has been handed to API Development team to review before implementation.

**Comment by GeoffLyle** [comment]
@jknable 
I'm seeing for this CDE [6347743](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6347743%20and%20ver_nr=1) that the permissible values **Poly-T Enrichment**, **Oligo-dT**, and **PolyA** are still present.  At the April 19th meeting I believe you stated that work had been done to change **Poly-T Enrichment** to **Poly-A Enriched Genomic Library**.  Could you further elaborate on that?  Perhaps I am looking at an out-of-date version of this CDE.

**Reply by jknable** [reply]
This CDE was versioned since we removed permissible values.  Please look at version 2 of the CDE.  The PVs are attached for your reference in my email response to this item.  I will try to attach here as well.
[CDE 6347743v2 PVs.xlsx](https://github.com/CBIIT/ccdi-federation-api/files/15119796/CDE.6347743v2.PVs.xlsx)
.

**Reply by GeoffLyle** [reply]
Ah, I see I was linking to version 1.
Updated link to CDE [6347742 v2](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6347743%20and%20ver_nr=2)

**Reply by GeoffLyle** [reply]
@jknable 
I see in the new version [6347742 v2](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6347743%20and%20ver_nr=2), that the change to **Poly-A Enriched Genomic Library** and the dropping of **Oligo-dT** and **PolyA** has been done.  I believe this is sufficient for an initial implementation (post v1 given the deadline).  There was also a request to remove the values that in excess to what the Federation members are currently using (see comment https://github.com/CBIIT/ccdi-federation-api/discussions/47#discussioncomment-8906307).  This would help Federation members and users better navigate the permissible values available.  After your review, can we still work to make these changes for the version 2 of this CDE?

**Reply by jknable** [reply]
Since this CDE is used by the broader caDSR community we can not limit the values displayed to your shorter list.  Groups typically handle this by restricting the allowable permissible values on their data collection template or case report form.

**Comment by hbeale** [comment]
Based on [the usage spreadsheet from march](https://docs.google.com/spreadsheets/d/1eT2ZahJTUq0ZihhVouSSqKQCYO0z2KrFDbgNB06DC2w/edit#gid=0), the following CDE 6347743 v2 values will be used by Federation members: Hybrid Selection, Not applicable, PCR, Poly-A Enriched Genomic Library, Random, rRNA Depletion and Unspecified. 

I've recorded the updated definitions in [a tab in the same spreadsheet labeled "v2"](https://docs.google.com/spreadsheets/d/1eT2ZahJTUq0ZihhVouSSqKQCYO0z2KrFDbgNB06DC2w/edit#gid=1101411284). The differences are: Removed treehouse's use of "other" per me, Removed CHOP's use of "Poly-T Enrichment" per [Hannah and Yuankun](https://uc-cri.slack.com/archives/C05FFPKLQ79/p1712158485603619?thread_ts=1711144987.319699&cid=C05FFPKLQ79), changed "PolyA" to "Poly-A Enriched Genomic Library" for all of us who report individual level data.

**Comment by GeoffLyle** [comment]
This CDE has been provisionally accepted at the June 21st Data Harmonization Meeting.  Federation API will use a subset of values in CDE [6347743 v2](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6347743%20and%20ver_nr=2).  These values are marked as "Used" in the [spreadsheet](https://docs.google.com/spreadsheets/d/1eT2ZahJTUq0ZihhVouSSqKQCYO0z2KrFDbgNB06DC2w/edit#gid=1101411284) reviewed by Federation members:

- Hybrid Selection
- Not applicable
- ~~PCR~~
- Poly-A Enriched Genomic Library
- Random
- Random PCR
- rRNA Depletion
- Unspecified

**Reply by Ssandor13** [reply]
@GeoffLyle after final discussion with St. Jude, we are mapping to `Random PCR` not `PCR`.
